src/shared.ts

```diff
@@ -224,66 +219,76 @@ api.storage.local.onChanged.addListener(items => {
 	api.storage.sync.get(DefaultOptions).then(options => {
 		const config = options as Config;
 		switch (e.type) {
-			case MessageEvent:
-				if (config.showBlockPopups) {
-					MakeToast(e.message, config, e.options);
-				}
-				break;
-
-			case UserBlockedEvent:
-				if (config.showBlockPopups) {
-					const event = e as BlockUser;
-					const { user, user_id, reason } = event;
-					const name =
-						user.name.length > 25
-							? user.name.substring(0, 23).trim() + '...'
-							: user.name;
-					const b = document.createElement('button');
-					b.innerText = 'undo';
-					b.onclick = () => {
-						unblockUser(user, user_id, reason);
-						const parent = b.parentNode as ParentNode;
-						parent.removeChild(b);
-					};
+		case MessageEvent:
+			if (config.showBlockPopups) {
+				MakeToast(e.message, config, e.options);
+			}
+			break;
+
+		case UserBlockedEvent:
+			if (config.showBlockPopups) {
+				const event = e as BlockUser;
+				const { user, user_id, reason } = event;
+				const name =
+					user.name.length > 25
+						? user.name.substring(0, 23).trim() + '...'
+						: user.name;
+				const b = document.createElement('button');
+				b.innerText = 'undo';
+				b.onclick = () => {
+					unblockUser(user, user_id, reason);
+					const parent = b.parentNode as ParentNode;
+					parent.removeChild(b);
+				};
+				const span = document.createElement("span");
+				span.appendChild(document.createTextNode(`${config.mute ? 'mut' : 'block'}ed ${EscapeHtml(name)} (`));
+				span.appendChild(((): HTMLAnchorElement => {
 					const screen_name = EscapeHtml(user.screen_name); // this shouldn't really do anything, but can't be too careful
-					MakeToast(
-						`${config.mute ? 'mut' : 'block'}ed ${EscapeHtml(
-							name,
-						)} (<a href="/${screen_name}">@${screen_name}</a>)`,
-						config,
-						{ html: true, elements: [b] },
-					);
-				}
-				break;
-
-			case UserLogoutEvent:
-				if (config.showBlockPopups) {
-					MakeToast(
-						`You have been logged out, and ${
-							config.mute ? 'mut' : 'block'
-						}ing has been paused.`,
-						config,
-						{
-							warn: true,
-						},
-					);
-				}
-				break;
+					const a = document.createElement("a");
+					a.href = "/" + screen_name;
+					a.innerText = "@" + screen_name;
+					return a;
+				})());
+				span.appendChild(document.createTextNode(")"));
+				MakeToast("", config, { elements: [span, b] });
+			}
+			break;
 
-			case ErrorEvent:
-				// skip checking options, since errors should always be shown
-				if (e.message) {
-					console.error(logstr, e.message, e);
-				}
+		case UserLogoutEvent:
+			if (config.showBlockPopups) {
 				MakeToast(
-					`<p>an error occurred! check the console and create an issue on <a href="https://github.com/kheina-com/Blue-Blocker/issues" target="_blank">GitHub</a></p>`,
+					`You have been logged out, and ${config.mute ? 'mut' : 'block'
+					}ing has been paused.`,
```

Comment:
Is BlueBlocker ever going to be localized? Constructing a string like this seems kinda bad for localization, but I guess that's ok if the add-on will only ever be English...

Reply:
no plans atm. I've thought about that too but haven't cared enough to make it happen lol

Reply:
all right